### PR TITLE
Move SplitControllerAndExtensions to Stable

### LINKED ISSUE #3513: Move SplitControllerAndExtensions to Stable
**Is your feature request related to a problem? Please describe.**
The "SplitControllerAndExtensions" feature has been in Beta since 1.32.0.

**Describe the solution you'd like**
Move it to stable, since it's been in Beta and turned on by default for a while.

**Describe alternatives you've considered**
Remove the functionality.

**Additional context**
Task list: https://github.com/googleforgames/agones/blob/065052a9beead20bbc2d7bf9a18a7906bc73a882/pkg/util/runtime/features.go#L93-L98

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3513 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Aha! Found it!

```
docker run --rm -v /workspace/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.config/helm:/root/.config/helm -v ~/.cache/helm:/root/.cache/helm -v /workspace:/go/src/agones.dev/agones -v /workspace/build//.gomod:/go/pkg/mod -v /workspace/build//.gocache:/root/.cache/go-build  agones-build:83141ac095 bash -c \
	'helm template agones-manual --namespace agones-system /go/src/agones.dev/agones/install/helm/agones \
	--set agones.controller.generateTLS=false \
	--set agones.allocator.generateTLS=false \
	--set agones.allocator.generateClientTLS=false \
	--set agones.crds.cleanupOnDelete=false \
	> /go/src/agones.dev/agones/install/yaml/install.yaml'
make[1]: Leaving directory '/workspace/build'
sort /workspace/install/yaml/install.yaml > /tmp/agones-install/install.current.yaml.sorted
diff /tmp/agones-install/install.yaml.sorted /tmp/agones-install/install.current.yaml.sorted
7a8,9
> 
> 
make: *** [Makefile:408: test-install-yaml] Error 1
make: *** Waiting for unfinished jobs....
```

Need to run `make gen-install` 😄

File: install/helm/agones/templates/controller.yaml [hunk 1]
```diff
@@ -12,11 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-{{- $featureGates := include "agones.featureGates" . | fromYaml }}
 {{- $replicas := (int .Values.agones.controller.replicas) }}
-{{- if not $featureGates.SplitControllerAndExtensions }}
-{{- $replicas = 1 }}
-{{- end }}
```
Comment (Kalaiselvi84) [inline install/helm/agones/templates/controller.yaml:19]: Is this correct? Do we need to keep {{- $replicas = 1 }}?

**Comment by Kalaiselvi84** [comment]
> Need to run `make gen-install` 😄

This command added two extra spaces. Could this be causing the build to fail?🤔

**Comment by markmandel** [comment]
> This command added two extra spaces. Could this be causing the build to fail?🤔

Yeah it will - any difference will cause it to fail 👍🏻

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -182,6 +183,45 @@ The following tables lists the configurable parameters of the Agones chart and t
...
 | `agones.controller.mutatingWebhook.disableCaBundle`      | Disable ca-bundle so it can be injected by cert-manager                                                                                                                                                                                                                                                          | `false`                                   |
 | `agones.controller.allocationBatchWaitTime`              | Wait time between each allocation batch when performing allocations in controller mode                                                                                                                                                                                                                           | `500ms`                                   |
+{{% /feature %}}
+
+{{% feature publishVersion="1.37.0" %}}
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:188]: Out of curiosity - what's the difference between this table and the above one?

**Review by markmandel — COMMENTED** [review]
Except for one question, this looks good to me. @zmerlynn can you also look over as well please, since you are more familiar with this work?

**Comment by zmerlynn** [comment]
Looks great! Sorry I missed this earlier.

**Review by zmerlynn — APPROVED** [review]

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 2 above)
Comment (zmerlynn) [reply site/content/en/docs/Installation/Install Agones/helm.md:188]: She took out the references to the feature gate, e.g. `agones.controller.replicas` previously had: `(Ignored unless [SplitControllerAndExtensions][split-controller] is enabled)`

Comment (markmandel) [reply site/content/en/docs/Installation/Install Agones/helm.md:188]: Ah thanks! I didn't see it 👍🏻

Comment (Kalaiselvi84) [reply site/content/en/docs/Installation/Install Agones/helm.md:188]: We have removed '(Ignored unless [SplitControllerAndExtensions][split-controller] is enabled)' from the description of a few parameters.

**Comment by markmandel** [comment]
Oooh, we got some link failures! You can run `make lint` to test locally.

**Comment by Kalaiselvi84** [comment]
> Oooh, we got some link failures! You can run `make lint` to test locally.

Fixed it. Something messed up when resolving the conflict.

**Pull request merged**